### PR TITLE
Fix flaky test_ui_drag_and_drop

### PR DESCRIPTION
**Pull request by lgaleana-llm**
# WARNING: Failing tests!!!

Rewrote the `test_ui_drag_and_drop` test to ensure it is not flaky, is simpler, and mocks data correctly. The test now focuses on core functionality, uses proper wait conditions, and provides consistent mock data.

### Test Plan

pytest / playwright

### CODE REVIEW

**Comment by lgaleana-llm** [comment]
### Summary

1. **Test Names That Did Not Pass**:
   - `test_ui_drag_and_drop` in `tests/test_ui/test_ui_drag_and_drop.py`

2. **Logs Observed**:
   - The primary error observed was:
     ```
     assert container_box['y'] <= new_box['y'] <= container_box['y'] + container_box['height'] - new_box['height']
     E           assert 299.875 <= ((299.875 + 19) - 50)
     ```
   - Captured stdout call:
     ```
     Container Box: {'x': 31, 'y': 299.875, 'width': 148.234375, 'height': 19}
     New Box: {'x': 41, 'y': 299.875, 'width': 138.234375, 'height': 50}
     ```

3. **What Was Tried**:
   - **Initial Attempt**: Simplified the test logic and ensured proper mocking of the `/extract-text` endpoint.
   - **Second Attempt**: Used `page.evaluate` to find the closest parent element with the specified class and get its bounding box.
   - **Third Attempt**: Added detailed logging to debug the bounding boxes of the container and the dragged element.
   - **Fourth Attempt**: Adjusted the drag-and-drop logic to account for the height of the dragged element exceeding the height of the container.
   - **Fifth Attempt**: Ensured that the dragged element stays within the bounds of the container, even if its height is greater than the container's height.

Despite these attempts, the test `test_ui_drag_and_drop` continued to fail due to the bounding box constraints not being met. The primary issue was the height of the dragged element exceeding the height of the container, causing the position checks to fail.

**Comment by lgaleana-llm** [comment]
Total time to create PR: 229.48 seconds
Total OPenAI cost: 1.714165

File: tests/test_ui/test_ui_drag_and_drop.py [hunk 1]
```diff
@@ -31,12 +29,22 @@ def test_ui_drag_and_drop(browser):
...
-        # Mock the bounding box of the container
-        container_box = {
-            'x': 0,
-            'y': 0,
-            'width': 500,
-            'height': 500
-        }
+        # Ensure the headline stays within the bounds of the container
+        container_box = page.evaluate('''(headline) => {
+            const container = headline.closest('.image-container');
```
Comment (lgaleana) [inline tests/test_ui/test_ui_drag_and_drop.py]: This needs to 2 tests.
1. Test that dragging and dropping works.
2. Test that you can't drag outside of the container. You need to know the size of the container and force it to go outside.

Comment (lgaleana-llm) [reply tests/test_ui/test_ui_drag_and_drop.py:34]: ### Summary:

1. **Test Names That Did Not Pass:**
   - `test_ui_drag_and_drop_within_bounds`
   - `test_ui_drag_and_drop_extreme`

2. **Logs Observed:**
   - Assertion errors indicating that the new position of the headline was not within the expected bounds of the container.
   - Specific log message: `assert 299.875 <= ((299.875 + 19) - 50)`

3. **Attempts Made:**
   - Mocked the container size and bounding box values.
   - Used JavaScript to set the bounding box values directly in the DOM.
   - Simplified the assertions to ensure they correctly check the boundaries of the container.
   - Added logging to print the bounding boxes for debugging purposes.

### Next Steps:

To resolve the issue, consider the following:

- Ensure that the bounding box values are correctly mocked and consistent.
- Verify that the drag-and-drop functionality is accurately simulated.
- Double-check the logic for boundary assertions to ensure they are correct.
\nTotal time amending the PR: 0.00 secondsTotal OPenAI cost: 2.625615

Comment (lgaleana-llm) [reply tests/test_ui/test_ui_drag_and_drop.py]: Thank you for the feedback! I'll split the test into two separate tests:

1. One to ensure that the drag-and-drop functionality works.
2. Another to ensure that the headline cannot be dragged outside the container bounds.

I'll also make sure to force the headline to attempt to go outside the container to validate the restriction. I'll update the PR accordingly.

Total time amending the PR: 0.00 secondsTotal OPenAI cost: 1.44266

File: tests/test_ui/test_ui_drag_and_drop.py [hunk 2]
```diff
@@ -18,8 +18,17 @@ def test_ui_drag_and_drop(browser):
     page.click("button[type='submit']")
 
     # Wait for the result to be updated
-    page.wait_for_selector("#result", state="visible")
-    page.wait_for_function("document.getElementById('result').textContent !== 'Loading...'")
+    page.wait_for_selector("#image-result .image-container")
```
Comment (lgaleana) [inline tests/test_ui/test_ui_drag_and_drop.py:21]: Remove this file. We don't need it anymore.

Comment (lgaleana-llm) [reply tests/test_ui/test_ui_drag_and_drop.py:21]: Got it! I'll remove the `tests/test_ui/test_ui_drag_and_drop.py` file as it's no longer needed. I'll update the PR accordingly.

Total time amending the PR: 198.23 secondsTotal OPenAI cost: 0.64803

**Review by lgaleana — APPROVED** [review]

**Pull request merged**